plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiTableHandle.java

```diff
@@ -37,49 +43,78 @@ public class HudiTableHandle
     private final String tableName;
     private final String basePath;
     private final HoodieTableType tableType;
-    private final Optional<String> preCombineField;
     private final List<HiveColumnHandle> partitionColumns;
     // Used only for validation when config property hudi.query-partition-filter-required is enabled
     private final Set<HiveColumnHandle> constraintColumns;
     private final TupleDomain<HiveColumnHandle> partitionPredicates;
     private final TupleDomain<HiveColumnHandle> regularPredicates;
+    // Coordinator-only
+    private final Optional<Table> table;
+    private final Optional<Lazy<HoodieTableMetaClient>> lazyMetaClient;
```

Comment:
Should we mark these as transient if they are not SerDe to the workers? i.e. HudiSplitSource receives a TableHandle too, but is not using this lazyMetaClient.

Also, would it be possible for us to SerDe the entire metaclient to workers?

Reply:
Hmm, if `HudiTableHandle` is not able to handle complex objects like metaClient, we might need to write a separate class ourselves.